### ISSUE TITLE
Exclude directories when deploying

### ISSUE THREAD

**Issue by deshartman** [twilio-labs/serverless-toolkit#301]
Hi,

Is there a way to exclude directories when using deploy, to avoid a "/src" directory being uploaded for example?

I am using the "/src" directory to build out static asset files, which I build into the assets folder, but want to avoid the code also being uploaded to Functions.

**Comment by philnash** [comment]
Hey @deshartman, the `src` directory was previously used by `twilio-run` as the directory to store the functions in. We have moved to using the `functions` directory by default now, but the legacy behaviour remains for deploys.

You have a few options here.

* If you don't have any functions to upload you can pass the flag `--no-functions`
* If you are using a different functions directory, specify it with the `--functions-folder`
* If you don't mind changing the directory name, using any directory not called `src` or `functions` will not get automatically deployed as functions.

Let me know if any of those work for you.

**Comment by deshartman** [comment]
@philnash , thanks for the reply and options, but it does not really solve it. The particular use case is if you have a project that has backend (functions) and frontend code. The "src" directory is used as the development environment, which you then build into the assets folder, so ideally you should exclude the src directory.

I guess the alternative is to develop the frontend in a different folder and build into the Functions assets folder and then deploy. Was just looking for a neat single approach.

FR: Would it be possible to add an "--exclude" flag or a ".ignore" file, which would then explicitly exclude particular directories from being uploaded to Functions? That seems like a useful addition to cover various use cases

**Comment by dkundel** [comment]
Hey @deshartman do you have a sample folder structure?

In theory a combination of the flags that Phil provided should work but I might also not understand your folder structure correctly.

Having a separate project directory inside your repository might also be an option that we see a lot for folks who build Flex plugins and Function backends in the same repository. Specifically because the dependencies on the package.json will be installed in your deployment.

If you could provide us with an example of your folder structure that would be helpful (no need for actual code in this case).

**Comment by deshartman** [comment]
@dkundel spoke to @philnash and we believe we found the issue. I still have to 100% confirm, but it was on my side. I had moved the src directory into the assets folder to test the deploy.

What this inadvertently uncovered is that not deleting older files, may be corrupting Functions, since we have older files just hanging around. To get around this, I deleted the Function and then ran deploy again.
